homeassistant/components/sems_portal/__init__.py

```diff
@@ -0,0 +1,107 @@
+"""The SEMS Portal integration."""
+from __future__ import annotations
+
+from asyncio import timeout
+from datetime import timedelta
+import logging
+from typing import Any
+
+from aiohttp import ClientError, ClientResponseError, ClientSession
+from sems_portal_api import get_collated_plant_details
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_EMAIL, CONF_POWERSTATIONID, CONF_TOKEN, DOMAIN, MANUFACTURER
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up SEMS Portal from a config entry."""
+
+    email: str = entry.data[CONF_EMAIL]
+    token: str = entry.data[CONF_TOKEN]
+    powerstationId: str = entry.data[CONF_POWERSTATIONID]
+    websession = async_get_clientsession(hass)
+
+    coordinator = SemsDataUpdateCoordinator(
+        hass, websession, email, token, powerstationId, email
+    )
+    await coordinator.async_config_entry_first_refresh()
+
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
```

Comment:
What do you use that update_listener for? You don't seem to have an options flow

Reply:
Thanks for that, that was some left over from a copy paste